modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectFactoryClassRepository.java

```diff
@@ -54,21 +55,26 @@
    * @param objectFactoryType the {@link ObjectFactory} of the component
    * @param createdObjectType the type of object created by the {@code ObjectFactory}
    * @param isLazyInitFunction function that defines if the object created by the component can be created lazily
-   * @param instancePostCreationFunction function to do custom processing of the created instance by the {@code ObjectFactory}.
+   * @param instancePostCreationFunctionOptional function to do custom processing of the created instance by the {@code ObjectFactory}.
    *        When there's no need for post processing this value must be {@link Optional#empty()}
    * @return the {@code FactoryBean} class to be used by spring for the provided configuration.
    */
   public Class<ObjectFactory> getObjectFactoryClass(ComponentBuildingDefinition componentBuildingDefinition,
                                                     Class objectFactoryType,
                                                     Class createdObjectType,
                                                     Supplier<Boolean> isLazyInitFunction,
-                                                    Consumer<Object> instancePostCreationFunction) {
+                                                    Optional<Consumer<Object>> instancePostCreationFunctionOptional) {
     try {
-      return objectFactoryClassCache.get(componentBuildingDefinition,
-                                         () -> getObjectFactoryDynamicClass(componentBuildingDefinition,
-                                                                            objectFactoryType,
-                                                                            createdObjectType, isLazyInitFunction,
-                                                                            instancePostCreationFunction));
+      if (instancePostCreationFunctionOptional.isPresent()) {
```

Comment:
use `map`/`orElse` instead of `isPresent`

Reply:
Tried that but since objectFactoryClassCache.get() declares that it throws an ExecutionException, the code needed to make it work with map/orElse would be worst than the one we have now.